src/ilasm/writer.cpp

```diff
@@ -982,6 +982,61 @@ BOOL Assembler::EmitEventsProps(Class* pClass)
     return ret;
 }
 
+HRESULT Assembler::AllocateStrongNameSignature()
+{
+    HRESULT             hr = S_OK;
+    HCEESECTION         hSection;
+    DWORD               dwDataLength;
+    DWORD               dwDataOffset;
+    DWORD               dwDataRVA;
+    VOID               *pvBuffer;
+    AsmManStrongName   *pSN = &m_pManifest->m_sStrongName;
+
+    // pSN->m_cbPublicKey is the length of the m_pbPublicKey
+    int keySize = (int)pSN->m_cbPublicKey;
+
+    if (keySize == 0)
+    {
+        dwDataLength = 0;
+    }
+    else
+    {
+        dwDataLength = (keySize < 128 + 32) ? 128 : keySize - 32;
+    }
+
+    // Grab memory in the section for our stuff.
+    if (FAILED(hr = m_pCeeFileGen->GetIlSection(m_pCeeFile,
+                                                &hSection)))
```

Comment:
Only thing I would ask is that you stop eliding braces.  I realize the formatting of this code is awful, but please just add braces in all of the code you added.